File: 03-09-2025_Find_No_of_ways_to_place_people_2.java
import java.util.*;

class Solution {
    public int numberOfPairs(int[][] nums) {
        // Sort by x ascending, if tie then y descending
        Arrays.sort(nums, (a, b) -> {
            if (a[0] == b[0]) return b[1] - a[1];
            return a[0] - b[0];
        });

        int n = nums.length;
        int count = 0;

        for (int i = 0; i < n; i++) {
            int maxi = Integer.MIN_VALUE;
            for (int j = i + 1; j < n; j++) {
                if (nums[j][1] > nums[i][1]) continue;
                if (nums[j][1] > maxi) {
                    count++;
                    maxi = nums[j][1];
                }
            }
        }
        return count;
    }
}

File: 03-09-2025_Find_No_of_ways_to_place_people_2.py
from typing import List

class Solution:
    def numberOfPairs(self, nums: List[List[int]]) -> int:
        # Sort by x ascending, if tie then y descending
        nums.sort(key=lambda x: (x[0], -x[1]))
        
        n = len(nums)
        count = 0

        for i in range(n):
            maxi = float("-inf")
            for j in range(i + 1, n):
                if nums[j][1] > nums[i][1]:
                    continue
                if nums[j][1] > maxi:
                    count += 1
                    maxi = nums[j][1]
        return count
